android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonFormFragment.java

```diff
@@ -271,6 +271,146 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * Skips blank by relevance steps when next is clicked on the json wizard forms.
+     */
+    public void skipLoadedStepsOnNextPressed() {
+        if (skipBlankSteps()) {
+            JSONObject formStep = getStep(getArguments().getString(JsonFormConstants.STEPNAME));
+            String next = formStep.optString(JsonFormConstants.NEXT, "");
+            if (StringUtils.isNotEmpty(next)) {
+                checkIfStepIsBlank(formStep);
+                if (shouldSkipStep() && !stepHasNoSkipLogic(JsonFormConstants.STEP1)) {
+                    getJsonApi().setNextStep(next);
+                    markStepAsSkipped(formStep);
+                    next();
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the current form step number when given than steps next step number.
+     * This number is used to figure out which steps to pop when previous is clicked.
+     *
+     * @param nextFormNumber {@link String}
+     * @return formNumber {@link Integer}
+     */
+    private int getFormStepNumber(String nextFormNumber) {
+        int formNumber = 0;
+        if (StringUtils.isNotBlank(nextFormNumber)) {
+            int currentFormNumber = Integer.parseInt(nextFormNumber.substring(4, 5)) - 1;
+            if (currentFormNumber > 0) {
+                formNumber = currentFormNumber;
+            } else if (currentFormNumber == 0) {
+                formNumber = 1;
+            }
```

Comment:
Can this be simplified to `formNumber = currentFormNumber > 0 ? currentFormNumber : 1` since a step is not expected to ever be negative?